api/src/pcapi/core/finance/api.py

```diff
@@ -2924,7 +2924,18 @@ def _has_been_recredited(user: users_models.User) -> bool:
     if len(user.deposit.recredits) == 0:
         return False
 
-    return conf.RECREDIT_TYPE_AGE_MAPPING[user.age] in [recredit.recreditType for recredit in user.deposit.recredits]
+    has_been_recredited = conf.RECREDIT_TYPE_AGE_MAPPING[user.age] in [
+        recredit.recreditType for recredit in user.deposit.recredits
+    ]
+    if has_been_recredited:
+        return True
+
+    if user.age == 16:
+        # when a user used to register at 15 and finishes their credit activation at 17, then the recredit 16 is applied
+        # to the deposit but the recredit 16 is not created in the database. see commit e9b5f5e4.
+        return user.deposit.amount == sum(conf.RECREDIT_TYPE_AMOUNT_MAPPING.values())
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏je comprend maintenant, parce qu'on en a parlé, je sais pas si c'est suffisamment clair pour un futur moi dans 6 mois

Proposition : 
```python
# This condition handles the following edge case: 
# - User started the activation workflow at 15 and finished it at 17. Result: 1 deposit holding the total amount from 15 to 17yo
# - However, when trying to unlock their 18yo deposit, we get a new (correct) value for their age
# - If this age is 16, no recredit happened (see 1st point) so we need a different check
# 
# This edge case is unlikely, but happened a few times already
```